### ISSUE TITLE
prune error about non-existenent resources

### ISSUE THREAD

**Issue by sdake** [istio/istio#19184]
**Bug description**
prune spits out a bunch of errors for deprecated resources

**Expected behavior**
Istio should not be using beta1 deployment nor app resources:

```
2019-11-24T22:13:57.359834Z     info    resource updated: admissionregistration.k8s.io/v1beta1, Kind=MutatingWebhookConfiguration
2019-11-24T22:13:59.151631Z     info    begin pruning
2019-11-24T22:13:59.942026Z     warn    retrieving resources to prune type apps/v1beta1, Kind=Deployment: no matches for kind "Deployment" in version "apps/v1beta1" not found
2019-11-24T22:14:02.148191Z     warn    retrieving resources to prune type apps/v1beta1, Kind=StatefulSet: no matches for kind "StatefulSet" in version "apps/v1beta1" not found
2019-11-24T22:14:04.349192Z     warn    retrieving resources to prune type extensions/v1beta1, Kind=DaemonSet: no matches for kind "DaemonSet" in version "extensions/v1beta1" not found
2019-11-24T22:14:06.542572Z     warn    retrieving resources to prune type extensions/v1beta1, Kind=Deployment: no matches for kind "Deployment" in version "extensions/v1beta1" not found
2019-11-24T22:14:08.743831Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=Certificate: no matches for kind "Certificate" in version "certmanager.k8s.io/v1beta1" not found
2019-11-24T22:14:10.941968Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=Challenge: no matches for kind "Challenge" in version "certmanager.k8s.io/v1beta1" not found
2019-11-24T22:14:13.142050Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=Issuer: no matches for kind "Issuer" in version "certmanager.k8s.io/v1beta1" not found
2019-11-24T22:14:15.341872Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=Order: no matches for kind "Order" in version "certmanager.k8s.io/v1beta1" not found
2019-11-24T22:14:17.541952Z     warn    retrieving resources to prune type rbac.istio.io/v1alpha1, Kind=AuthorizationPolicy: no matches for kind "AuthorizationPolicy" in version "rbac.istio.io/v1alpha1" not found
2019-11-24T22:14:19.742552Z     warn    retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=ClusterIssuer: no matches for kind "ClusterIssuer" in version "certmanager.k8s.io/v1beta1" not found
2019-11-24T22:14:20.120724Z     info    end pruning
2019-11-24T22:14:20.143136Z     info    end reconciling resources
```
**Steps to reproduce the bug**
IKS on 1.16 - deploy operator and viola the logs wlil reproduce for the operator.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
istio 1.1.4 IKS 1.16.3.

**How was Istio installed?**
ICP standalone operator

**Environment where bug was observed (cloud vendor, OS, etc)**
IBM Cloud

**Comment by sdake** [comment]
I'm speculating this is cert manager not being deployable resulting in a prune action because the controller belives these cert manager resources have been deployed.

**Comment by howardjohn** [comment]
I think there is just a list of stuff it tries to prune, and if they don't exist (like if cert manager is not installed) then you get this error

**Comment by sdake** [comment]
There are a few error conditions with prune I can think of off the top of my head:

1. The operator has things to prune, however, the prune fails in progress and should be completed on the next reconciliation.
1. The operator does not have this resource as a deployed resource prior, yet there are resources found indicative of duplicate resources deployed.
1. The operator has things to prune, but the prune fails to get started (which is similar to 1, but a subset).

In each of these cases, the operator should be printing different info level messages about why the prune is taking place.  At present, an error prune shows up in a grep of the logs as an error - which is clearly not desirable :)

Cheers
-steve
That second case

**Comment by sdake** [comment]
@howardjohn - to be more direct, a spew of 8 messages at WARN level that appear as if something went wrong, while the actual results are indeterminate is not an acceptable log message nor solution.  Even now, it is not possible to detect if this was a prune error on the human side, a prune error in our implementation, or a prune error in the infrastructure.

Cheers
-steve

Cheers
-steve

**Comment by howardjohn** [comment]
Yeah something is definitely wrong, just explaining the root cause. Those resources do not exist on your cluster, the k8s ones are because those were removed in k8s 1.16

**Comment by howardjohn** [comment]
@ostromart I think this is fixed, can yo confirm?

**Comment by lee-zhg** [comment]
Hi, is there a solution/workaround for this error? I hit the same problem in Istioctl v1.5.2. 

It appears there are .yaml file(s) somewhere that refers "apiVersion: extensions/v1beta1".

istioctl manifest apply --set profile=demo

- Applying manifest for component Base...
✔ Finished applying manifest for component Base.
- Applying manifest for component Pilot...
✔ Finished applying manifest for component Pilot.
- Applying manifest for component EgressGateways...
- Applying manifest for component IngressGateways...
- Applying manifest for component AddonComponents...
2020-04-30T22:32:51.915154Z	error	installer	error running kubectl: exit status 1
2020-04-30T22:32:59.390184Z	error	installer	error running kubectl: exit status 1
✘ Finished applying manifest for component EgressGateways.
2020-04-30T22:33:00.453259Z	error	installer	error running kubectl: exit status 1
✘ Finished applying manifest for component IngressGateways.
2020-04-30T22:33:03.435010Z	error	installer	error running kubectl: exit status 1
✘ Finished applying manifest for component AddonComponents.

Component EgressGateways - manifest apply returned the following errors:
Error: error running kubectl: exit status 1

Error detail:

error: error retrieving RESTMappings to prune: invalid resource extensions/v1beta1, Kind=DaemonSet, Namespaced=true: no matches for kind "DaemonSet" in version "extensions/v1beta1" (repeated 1 times)


serviceaccount/istio-egressgateway-service-account created
deployment.apps/istio-egressgateway created
poddisruptionbudget.policy/istio-egressgateway created
service/istio-egressgateway created



Component IngressGateways - manifest apply returned the following errors:
Error: error running kubectl: exit status 1

Error detail:

error: error retrieving RESTMappings to prune: invalid resource extensions/v1beta1, Kind=DaemonSet, Namespaced=true: no matches for kind "DaemonSet" in version "extensions/v1beta1" (repeated 1 times)


serviceaccount/istio-ingressgateway-service-account created
gateway.networking.istio.io/istio-ingressgateway created
sidecar.networking.istio.io/default created
deployment.apps/istio-ingressgateway created
poddisruptionbudget.policy/ingressgateway created
role.rbac.authorization.k8s.io/istio-ingressgateway-sds created
rolebinding.rbac.authorization.k8s.io/istio-ingressgateway-sds created
service/istio-ingressgateway created



Component AddonComponents - manifest apply returned the following errors:
Error: error running kubectl: exit status 1

Error detail:

error: error retrieving RESTMappings to prune: invalid resource extensions/v1beta1, Kind=DaemonSet, Namespaced=true: no matches for kind "DaemonSet" in version "extensions/v1beta1" (repeated 1 times)

error: error retrieving RESTMappings to prune: invalid resource extensions/v1beta1, Kind=DaemonSet, Namespaced=true: no matches for kind "DaemonSet" in version "extensions/v1beta1" (repeated 1 times)

**Comment by howardjohn** [comment]
This has been fixed in master/1.6 now
